src/uct/ib/ud/base/ud_ep.c

```diff
@@ -1704,3 +1704,29 @@ void uct_ud_ep_disconnect(uct_ep_h tl_ep)
 
     uct_ud_leave(iface);
 }
+
+#if UCT_UD_EP_DEBUG_HOOKS
+
+static inline ucs_status_t
+uct_ud_ep_invalid_timer_hook(uct_ud_ep_t *ep, uct_ud_neth_t *neth)
+{
+    ep->tx.send_time = 0;
+    return UCS_OK;
+}
+
+ucs_status_t uct_ud_ep_invalidate(uct_ep_h tl_ep, unsigned flags)
+{
+    uct_ud_ep_t *ep = ucs_derived_of(tl_ep, uct_ud_ep_t);
+
+    ep->timer_hook = uct_ud_ep_invalid_timer_hook;
```

Comment:
why not just set ep->tx.send_time = 0 here?
then no need ifdef

Reply:
every post updates `send_time`

Reply:
then err flow will be significantly changed

Reply:
check why calling same code as https://github.com/openucx/ucx/blob/45b89935f8770e2d25c4a1d8ab1cfd08b6831800/src/uct/ib/ud/base/ud_ep.c#L318-L325 fails